### PR TITLE
some ngraph updates to enable bert

### PR DESCRIPTION
**Pull request by baojun-nervana**
This PR includes a number of changes to enable BERT through ngraph:
1. set ngraph to support block 0 only, the subblock from condition_block op will run through paddle
2. Infershape was delayed before getting output dimension. 
    - reason for this change is some of the variables are reused in some reshape ops and it will impact the inputs when the inputs trying to pull the shape.
    - mul_op and pool2d were updated accordingly per this change
3. fix some corner cases
    - handle shape {0}
    - partial support of lookup_table_grad
4. remove duplicated output
   - this is an optimization
5. Handle reshape2_grad infershape
    Paddle reshape2 add a leading dimension 0 and chop it off in reshape_grad infershape. Ngraph would interpret 0 as 0 element, so it needs special handling in ngraph engine

### CODE REVIEW

**Review by mozga-intel — APPROVED** [review]
LGTM!

File: paddle/fluid/operators/ngraph/ngraph_engine.cc [hunk 1]
```diff
@@ -626,6 +631,21 @@ void NgraphEngine::Run(const framework::Scope& scope,
     }
   }
 
+  for (auto& op : fused_ops_) {
+    framework::RuntimeContext ctx(op->Inputs(), op->Outputs(), scope_);
+    if (op->Type() == "reshape2_grad") {
```
Comment (tensor-tang) [inline paddle/fluid/operators/ngraph/ngraph_engine.cc:636]: why check reshape2 type in `Run` ? what if we have more ? transpose2 or others?

Comment (baojun-nervana) [reply paddle/fluid/operators/ngraph/ngraph_engine.cc:636]: reshape op adds a leading dimension of 0. https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/reshape_op.cc#L261

And chop it off in its grad op, https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/reshape_op.cc#L311.

In paddle the dimension 0 means placeholder or removable dimension, however ngraph will check dimension product and 0 will make ngraph treat the tensor as 0 element. So we have to process reshape2_grad op infershape differently. Doing this in ngraph engine can avoid modifying the original paddle file.

**Comment by LeoZhao-BJ** [comment]
LGTM

**Review by tensor-tang — APPROVED** [review]
LGTM

**Pull request merged**